src/credentials.c

```diff
@@ -7,37 +7,73 @@
 #include <aws/auth/credentials.h>
 #include <aws/common/string.h>
 
-struct aws_credentials *aws_crt_credentials_new(
-    const char *access_key_id, const char *secret_access_key,
-    const char *session_token, uint64_t expiration_timepoint_seconds) {
+struct _aws_crt_credentials_options {
```

Comment:
credentials can hold an ecc key now too

Reply:
Publicly? I don't see this in auth/credentials.h

Reply:
Oh yeah, umm forget I said that.